AWS provider will not build against latest Terraform v0.12.0-dev modules due to error in r/aws_glue_crawler

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform Version

<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->

Terraform v0.12.0-dev

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* aws_glue_crawler

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

### Debug Output

<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

To obtain the debug output, see the [Terraform documentation on debugging](https://www.terraform.io/docs/internals/debugging.html).
--->

### Panic Output

<!--- If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`. --->

### Expected Behavior

<!--- What should have happened? --->

```console
$ go build
$
```

### Actual Behavior

<!--- What actually happened? --->

```console
$ go build
# github.com/terraform-providers/terraform-provider-aws/aws
aws/resource_aws_glue_crawler.go:369:47: cannot convert *crawlerInput (type glue.CreateCrawlerInput) to type glue.UpdateCrawlerInput
```

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

```console
$ cd $GOPATH/src/github.com/terraform-providers
$ git clone https://github.com/terraform-providers/terraform-provider-aws.git
$ cd terraform-provider-aws
$ go get -u github.com/hashicorp/terraform@16823f43d
$ go build
go: finding github.com/hashicorp/terraform v0.12.0-alpha4.0.20190225213247-16823f43deec
# github.com/terraform-providers/terraform-provider-aws/aws
aws/resource_aws_glue_crawler.go:369:47: cannot convert *crawlerInput (type glue.CreateCrawlerInput) to type glue.UpdateCrawlerInput
```

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? --->
Working with Terraform v0.12.0-dev and compiling AWS provider with latest go modules.

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor documentation? For example:
--->

--- a/aws/resource_aws_glue_crawler.go
+++ b/aws/resource_aws_glue_crawler.go
@@ -251,6 +251,51 @@ func createCrawlerInput(crawlerName string, d *schema.ResourceData) (*glue.Creat
 	return crawlerInput, nil
 }
 
+func updateCrawlerInput(crawlerName string, d *schema.ResourceData) (*glue.UpdateCrawlerInput, error) {
+	crawlerTargets, err := expandGlueCrawlerTargets(d)
+	if err != nil {
+		return nil, err
+	}
+	crawlerInput := &glue.UpdateCrawlerInput{
+		Name:         aws.String(crawlerName),
+		DatabaseName: aws.String(d.Get("database_name").(string)),
+		Role:         aws.String(d.Get("role").(string)),
+		Targets:      crawlerTargets,
+	}
+	if description, ok := d.GetOk("description"); ok {
+		crawlerInput.Description = aws.String(description.(string))
+	}
+	if schedule, ok := d.GetOk("schedule"); ok {
+		crawlerInput.Schedule = aws.String(schedule.(string))
+	}
+	if classifiers, ok := d.GetOk("classifiers"); ok {
+		crawlerInput.Classifiers = expandStringList(classifiers.([]interface{}))
+	}
+
+	crawlerInput.SchemaChangePolicy = expandGlueSchemaChangePolicy(d.Get("schema_change_policy").([]interface{}))
+
+	if tablePrefix, ok := d.GetOk("table_prefix"); ok {
+		crawlerInput.TablePrefix = aws.String(tablePrefix.(string))
+	}
+	if configuration, ok := d.GetOk("configuration"); ok {
+		crawlerInput.Configuration = aws.String(configuration.(string))
+	}
+
+	if v, ok := d.GetOk("configuration"); ok {
+		configuration, err := structure.NormalizeJsonString(v)
+		if err != nil {
+			return nil, fmt.Errorf("Configuration contains an invalid JSON: %v", err)
+		}
+		crawlerInput.Configuration = aws.String(configuration)
+	}
+
+	if securityConfiguration, ok := d.GetOk("security_configuration"); ok {
+		crawlerInput.CrawlerSecurityConfiguration = aws.String(securityConfiguration.(string))
+	}
+
+	return crawlerInput, nil
+}
+
 func expandGlueSchemaChangePolicy(v []interface{}) *glue.SchemaChangePolicy {
 	if len(v) == 0 {
 		return nil
@@ -362,15 +407,14 @@ func resourceAwsGlueCrawlerUpdate(d *schema.ResourceData, meta interface{}) erro
 	glueConn := meta.(*AWSClient).glueconn
 	name := d.Get("name").(string)
 
-	crawlerInput, err := createCrawlerInput(name, d)
+	updateCrawlerInput, err := updateCrawlerInput(name, d)
 	if err != nil {
 		return err
 	}
-	updateCrawlerInput := glue.UpdateCrawlerInput(*crawlerInput)
 
 	// Retry for IAM eventual consistency
 	err = resource.Retry(1*time.Minute, func() *resource.RetryError {
-		_, err := glueConn.UpdateCrawler(&updateCrawlerInput)
+		_, err := glueConn.UpdateCrawler(updateCrawlerInput)
 		if err != nil {
 			if isAWSErr(err, glue.ErrCodeInvalidInputException, "Service is unable to assume role") {
 				return resource.RetryableError(err)

